Generated client.ts import of ./types missing file extension

### Describe the issue that you're seeing. Any videos (eg. YouTube or Loom) or screenshots help!

If I try to import the `tina/__generated__/client.ts` file in a Node.js script (e.g. to run an express server for the datalayer), I get an error like this:
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/daniel/src/tina-selfhost-test1/tina/__generated__/types' imported from /home/daniel/src/tina-selfhost-test1/tina/__generated__/client.ts
```

According to the Node.js documentation, [file extensions must be provided](https://nodejs.org/api/esm.html#mandatory-file-extensions) when using ES modules, so this import should actually have a `.ts` extension, like this:
```js
import { queries } from "./types.ts";
```

not:
```js
import { queries } from "./types";
```

### Reproduction

N/A

### Steps to reproduce

Create new TinaCMS site (I used the Astro template)
Create new `server.ts` file:
```js
import {client} from "../../tina/__generated__/client.ts";
```

Run it:
```
node server.ts
```

### System Info

```shell
System:
    OS: Linux 6.15 Fedora Linux 42 (KDE Plasma Desktop Edition)
    CPU: (24) x64 AMD Ryzen 9 5900X 12-Core Processor
    Memory: 16.04 GB / 31.24 GB
    Container: Yes
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.21.0 - /home/daniel/.volta/tools/image/node/22.21.0/bin/node
    Yarn: 4.10.3 - /home/daniel/.volta/tools/image/yarn/4.10.3/bin/yarn
    npm: 10.9.4 - /home/daniel/.volta/tools/image/node/22.21.0/bin/npm
  Browsers:
    Firefox: 141.0
    Firefox Developer Edition: 141.0
  npmPackages:
    @tinacms/cli: ^1.11.0 => 1.11.0 
    @tinacms/datalayer: ^1.4.1 => 1.4.1 
    tinacms: ^2.9.0 => 2.9.0
```

### Validations

- [x] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [docs](https://tina.io/docs).
- [x] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

--- a/.changeset/fix-generated-client-esm-extension.md
+++ b/.changeset/fix-generated-client-esm-extension.md
@@ -1,0 +1,5 @@
+---
+"@tinacms/cli": patch
+---
+
+Fix missing .js extension on ./types import in generated client.ts and databaseClient.ts

--- a/packages/@tinacms/cli/src/next/codegen/index.test.ts
+++ b/packages/@tinacms/cli/src/next/codegen/index.test.ts
@@ -19,6 +19,58 @@ jest.mock('esbuild', () => ({
 import * as stripModule from './stripSearchTokenFromConfig';
 import { Codegen } from './index';
 
+describe('Codegen.genClient', () => {
+  function makeInstance(isTs: boolean): Codegen {
+    const instance = Object.create(Codegen.prototype) as Codegen;
+    instance.apiURL = 'https://example.com/graphql';
+    instance.localContentBuild = false;
+    instance.noClientBuildCache = true;
+    instance.configManager = {
+      config: { token: 'tok' },
+      generatedCachePath: '/fake/cache',
+      isUsingTs: () => isTs,
+    } as any;
+    return instance;
+  }
+
+  it('emits ./types.ts import for TypeScript projects', async () => {
+    const { clientString } = await makeInstance(true).genClient();
+    expect(clientString).toContain('from "./types.ts"');
+    expect(clientString).not.toMatch(/from ["']\.\/types["']/);
+  });
+
+  it('emits ./types.js import for non-TypeScript projects', async () => {
+    const { clientString } = await makeInstance(false).genClient();
+    expect(clientString).toContain('from "./types.js"');
+    expect(clientString).not.toMatch(/from ["']\.\/types["']/);
+  });
+});
+
+describe('Codegen.genDatabaseClient', () => {
+  function makeInstance(isTs: boolean): Codegen {
+    const instance = Object.create(Codegen.prototype) as Codegen;
+    instance.configManager = {
+      isUsingTs: () => isTs,
+    } as any;
+    instance.tinaSchema = {
+      getCollections: () => [],
+    } as any;
+    return instance;
+  }
+
+  it('emits ./types.ts import for TypeScript projects', async () => {
+    const result = await makeInstance(true).genDatabaseClient();
+    expect(result).toContain('from "./types.ts"');
+    expect(result).not.toMatch(/from ["']\.\/types["']/);
+  });
+
+  it('emits ./types.js import for non-TypeScript projects', async () => {
+    const result = await makeInstance(false).genDatabaseClient();
+    expect(result).toContain('from "./types.js"');
+    expect(result).not.toMatch(/from ["']\.\/types["']/);
+  });
+});
+
 describe('Codegen.execute integration', () => {
   let spy: jest.SpyInstance;
   const SAFE_RESULT = { branch: 'main', token: 'tok' };

--- a/packages/@tinacms/cli/src/next/codegen/index.ts
+++ b/packages/@tinacms/cli/src/next/codegen/index.ts
@@ -297,7 +297,7 @@ export class Codegen {
 import { resolve } from "@tinacms/datalayer";
 import type { TinaClient } from "tinacms/dist/client";
 
-import { queries } from "./types";
+import { queries } from "${this.configManager.isUsingTs() ? './types.ts' : './types.js'}";
 import database from "../database";
 
 export async function databaseRequest({ query, variables, user }) {
@@ -365,7 +365,7 @@ export default databaseClient;
     const apiURL = this.getApiURL();
 
     const clientString = `import { createClient } from "tinacms/dist/client";
-import { queries } from "./types";
+import { queries } from "${this.configManager.isUsingTs() ? './types.ts' : './types.js'}";
 export const client = createClient({ ${
       this.noClientBuildCache === false
         ? `cacheDir: '${normalizePath(

